go/libraries/doltcore/doltdb/doltdb.go

```diff
@@ -1280,7 +1280,7 @@ func (ddb *DoltDB) PullChunks(ctx context.Context, tempDir string, srcDB *DoltDB
 		puller, err := datas.NewPuller(ctx, tempDir, defaultChunksPerTF, srcDB.db, ddb.db, stRef.TargetHash(), pullerEventCh)
 
 		if err == datas.ErrDBUpToDate {
-			return nil
+			return datas.ErrDBUpToDate
```

Comment:
Changing this will change the behavior at its call sites. I would expect one or more tests to die from this change.

Also, this condition should just collapse to:

```go
if err != nil {
    return err
}
```